dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/inject/InjectInjectionBinding.java

```diff
@@ -122,17 +127,128 @@ public void cleanAfterMethodInvocation() {
         }
     }
 
+    /*
+     * (non-Javadoc)
+     * @see com.ibm.wsspi.injectionengine.InjectionBinding#getInjectionObject(java.lang.Object, com.ibm.wsspi.injectionengine.InjectionTargetContext)
+     * 
+     * This method was copied from InjectionBinding.getInjectionObject() and has three differences.
+     * 
+     * Firstly it does not look in the InjectionBinding.ivInjectedObject field, as InjectInjectionBinding never has that field set anyway.
+     *   
+     * Secondly if the property com.ibm.ws.cdi.ignoreInjectionFailure is true it will ignore any failure except a RecursiveInjectionException
+     * The special status of RecursiveInjectionException is inherited from InjectionBinding where RecursiveInjectionException was the only 
+     * Throwable that isn't converted into an InjectionException. 
+     * 
+     * Thirdly if getInjectionObjectInstance returns an InjectableNUll this method will turn it into a regular null after the null check.
+     * This allows us to impliment section 3.3. of the CDI spec which allows @Dependent scoped producers to return a null. 
+     */
     @Override
     public Object getInjectionObject(Object targetObject,
-                                     InjectionTargetContext targetContext) throws InjectionException {
-        try {
-            return super.getInjectionObject(targetObject, targetContext);
-        } catch (InjectionException e) {
-            if (!CDIUtils.isInjectionFailureIgnored()) {
-                throw e;
-            } else {
+            InjectionTargetContext targetContext)
+                    throws InjectionException {
+
+        Object retObj = null;
+
+        try
+        {
+            retObj = getInjectionObjectInstance(targetObject, targetContext);
+        } catch (RecursiveInjectionException ex)
+        {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled())
+                Tr.exit(tc, "getInjectionObject: " + ex);
+            throw ex;
+        } catch (Throwable ex)
+        {
+
+            if (CDIUtils.isInjectionFailureIgnored()){
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "com.ibm.ws.cdi.ignoreInjectionFailure = true, ignoring InjectionException", ex);
+                }
+                return null;
```

Comment:
There are five return statements in this method, can that be cut down a bit? Ideally just one at the end. Comments will help people understand the flow.